libcaf_core/caf/local_actor.hpp

```diff
@@ -264,6 +266,21 @@ class CAF_CORE_EXPORT local_actor : public abstract_actor {
     monitor(actor_cast<abstract_actor*>(whom), P);
   }
 
+  /// Adds a unidirectional `monitor` to `whom` with custom default handler.
+  /// @returns a disposable object that demonitors the actor on disposing.
+  /// @note This overload does not work with the @ref demonitor member function.
+  template <typename Handle, typename Fn>
+  disposable monitor(Handle whom, Fn func) {
+    using impl_t = detail::monitor_action<Fn>;
+    auto monitor_task = make_counted<impl_t>(std::move(func));
+    whom->attach_functor([self = this, whom, monitor_task](error reason) {
+      monitor_task->arg(down_msg{whom.address(), std::move(reason)});
```

Comment:
This should return `true` if the action is still pending and `false` otherwise. Then, we can check the result here and don't send the message if the action has already been disposed.